tracee-rules/engine/engine.go

```diff
@@ -13,6 +15,7 @@ type Engine struct {
 	logger          log.Logger
 	signatures      map[types.Signature]chan types.Event
 	signaturesIndex map[types.SignatureEventSelector][]types.Signature
+	signaturesMutex sync.RWMutex
```

Comment:
can you explain we we need a RWMutex here instead of the simpler Mutex?

Reply:
We will have multiple readers on the map, for example ListSignatures().